event/tests/tests_view.py

```diff
@@ -0,0 +1,86 @@
+# from django.test import TestCase
+from django.test import TestCase
+from event.models import Event
+from organisation.models import Organisation
+from user.models import User
+from django.utils import timezone
+from django.urls import reverse, reverse_lazy
+import json
+from .tests_config import TEST_USER_CREDENTIALS as test_user
+from django.forms.models import model_to_dict
+
+# Create your tests here.
+MODEL_FIELDS = ['id', 'title', 'description']
+
+class EventViewTest(TestCase):
+    @classmethod
+    def create_event(cls, title='test event', description='test desc'):
+        usr = User.objects.create(
+            username=test_user['username'], email=test_user['email'])
+        usr.set_password(test_user['password'])
+        usr.save()
+        org = Organisation.objects.create(name='test', owner=usr)
+        return Event.objects.create(title=title, description=description,
+                                    start_datetime=timezone.now(), end_datetime=timezone.now(), organisation=org)
+
+    def test_event_list_view(self):
+        event_instance = self.create_event()
+        url = reverse_lazy('event-list')
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)['results'].pop()
+        self.assertDictContainsSubset(model_to_dict(event_instance, MODEL_FIELDS), json_resp)
+
+    def test_event_detail_view(self):
+        event_instance = self.create_event()
+        url = reverse('event-detail', args=[event_instance.id])
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)
+        self.assertDictContainsSubset(model_to_dict(event_instance, MODEL_FIELDS), json_resp)
+
+    def test_event_create_view_without_token(self):
+        event_instance = self.create_event()
+        url = reverse('event-list')
+        resp = self.client.post(url, {'title': 'new event', 'description': 'testing',
+                                      'start_datetime': timezone.now(), 'end_datetime': timezone.now(), 'organisation': event_instance.organisation})
+        # 401 status code will be sent by the backend for unauthorized requests
+        self.assertEqual(resp.status_code, 401)
+
+    def test_event_edit_view_without_token(self):
+        event_instance = self.create_event()
+        url = reverse('event-detail', args=[event_instance.id])
+        data = json.dumps({'title': 'update event title'})
+        resp = self.client.patch(
+            url, data, content_type='application/json')
+        # 401 status code will be sent by the backend for unauthorized requests
+        self.assertEqual(resp.status_code, 401)
+
+    def test_event_create_view_with_token(self):
```

Comment:
the test permissions should be a testing all permissions e.g. delete and create as well?

Reply:
No same permissions applied with testing, for authorized requests we are sending a token along with request.

Reply:
@mashhoodr you mean we need to change permissions for testing?

Reply:
What I mean is instead of just calling one request, a single persmission test should call all endpoints which should return 401 and and assert that

Reply:
@mashhoodr Alright got your point.